### PR TITLE
8303472: Display name for region TR

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
This PR changes the English name for the region `TR`, from `Turkey` to `Türkiye`. Although this change is included in the upcoming CLDR v43, it should be applied as a spot change so that it can be back-ported properly (As it is a common English region name).

This change targets both the CLDR and COMPAT data.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8303472](https://bugs.openjdk.org/browse/JDK-8303472): Display name for region TR

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)
 * @srl295 (no known openjdk.org user name / role)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12816/head:pull/12816` \
`$ git checkout pull/12816`

Update a local copy of the PR: \
`$ git checkout pull/12816` \
`$ git pull https://git.openjdk.org/jdk pull/12816/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12816`

View PR using the GUI difftool: \
`$ git pr show -t 12816`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12816.diff">https://git.openjdk.org/jdk/pull/12816.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/sun/util/resources/LocaleNames.properties [hunk 1]
```diff
@@ -1094,7 +1094,7 @@ TL=Timor-Leste
 TM=Turkmenistan
 TN=Tunisia
 TO=Tonga
-TR=Turkey
+TR=T�rkiye
```
Comment (naotoj) [inline src/java.base/share/classes/sun/util/resources/LocaleNames.properties:1097]: I believe this is fine, but please confirm that this ends up correct u-umulaut when you get the text. It is somewhat a rare case where ISO-8859-1 has the character and the byte sequence differs between ISO-8859-1 and UTF-8.

**Review by naotoj — COMMENTED** [review]
Please supply test cases with these changes (`sun/text/resources/LocaleDataTest.java`), also copyright year changes for COMPAT properties.
Otherwise LGTM

File: src/java.base/share/classes/sun/util/resources/LocaleNames.properties (hunk 1 above)
Comment (justin-curtis-lu) [reply src/java.base/share/classes/sun/util/resources/LocaleNames.properties]: Thanks for pointing that out, I switched it to just be a unicode escaped ü, as that would be more consistent with the rest of the file. I also added the changes to the test files like you suggested.

**Review by naotoj — APPROVED** [review]

**Review by srl295 — APPROVED** [review]

**Pull request closed without merging**